src/main/java/net/dries007/tfc/common/blocks/CalciteBlock.java

```diff
@@ -92,7 +93,7 @@ public boolean canSurvive(BlockState state, IWorldReader worldIn, BlockPos pos)
     {
         BlockPos abovePos = pos.above();
         BlockState aboveState = worldIn.getBlockState(abovePos);
-        return (aboveState.getBlock() == this && !aboveState.getValue(TIP)) || aboveState.isFaceSturdy(worldIn, abovePos, Direction.DOWN);
+        return (aboveState.getBlock() == this && !aboveState.getValue(TIP)) || aboveState.is(BlockTags.BASE_STONE_OVERWORLD) || aboveState.is(BlockTags.ICE);
```

Comment:
This should probably stay with the validity check here (i.e. "is face sturdy") for calcite. If ice does not pass that check, it should probably be adjusted just for ice (e.g. calcite should not survive on ice. This also applies if we want to make icicles accumulate in winter storms (*which would be bloody cool!*). If this causes issues during world gen that should be handled by the feature imo.